### PR TITLE
ec2_elb_lb boto3 migration

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Rewrite ec2_elb_lb using the boto3 SDK

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ec2_elb_lb

##### ADDITIONAL INFORMATION

Current state

- [x] renamed to elb_classic_lb
- [x] create/delete ELBs
- [x] change schema
- [x] update availability-zones
- [x] update subnets
- [x] manage tags
- [x] manage listeners
- [x] manage health checks
- [x] manage cross-AZ balancing
- [x] manage draining
- [x] manage idle_timeout
- [x] manage backend policies
- [x] manage stickiness
- [x] manage Instance attachment
- [x] manage access logging
- [x] lookup SGs by name
- [x] parameter validation (protocol, mandatory on creation etc)

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1097

### CODE REVIEW

**Comment by tremble** [comment]
Current state
- can create/delete ELBs
- can't update
- can't manage tags
- can't manage health checks

**Comment by jillr** [comment]
following up on our irc convo, I'm +1 to ec2_classic_lb with redirects (the ec2 classic announcements read like classic ELBs will also be retired, but I'm skimming quickly so I could have missed something)

**Comment by tremble** [comment]
Still to do:

- 'complex' parameter Validation (things like required if creating)
- lookup SG by name

**Comment by tremble** [comment]
@jillr @alinabuzachis - the module still doesn't have output docs (wasn't there originally) but it's probably going to be a long review...

**Comment by tremble** [comment]
Re @jillr's EC2 Classic retirement comment...

https://docs.aws.amazon.com/elasticloadbalancing/latest/userguide/migrate-classic-load-balancer.html#migrate-step-by-step-classiclink

What's being retired is only ELB Classic LBs using EC2 Classic networking (ie killing off the EC2 Classic networking component).  Amazon's documentation includes "Option 4: Migrate manually to a Classic Load Balancer in a VPC" (They recommend migrating to an ALB/NLB as options 1-3).  This implies that VPC attached ELBs are staying.

NLBs are less than 4 years old, where as it wasn't even possible to create new EC2 Classic accounts 8 years ago (without explicitly requesting them).   I'd expect to see creation of new "ELB Classic" resources being blocked in some why before they try to kill off ELBs themselves.  At this point I'm guessing AWS are looking at orders of magnitude more ELBs compared to EC2 Classic Networking resources.  (Amazon's not even started using their usual trick of making it financially advantageous to migrate)

**Review by jillr — COMMENTED** [review]
I'm trying to use the new integration tests with the old module (to demonstrate the module functionality is retained) unfortunately there's a lot of return msg changes so I have to slowly comment my way through them (with check_mode tests already disabled).  I understand why this is the case but it is making reviewing slower and more challenging.  Still working on it though!

**Comment by tremble** [comment]
@jillr I've re-added the 'fact' (for what little a hard-coded value is worth).  As such we should be able to land this in a normal minor release rather than this needing to wait for a major release.

There are a couple of similar 'facts' over in community.aws that we should probably prune at the same time we clean out all of the _fact aliases, and just have 3.0 labeled as our big "fact" cleanup.

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
@goneri Something wierd's going on with the tests at the minute:
https://3eacf59dc8ee4ec2e19f-f4cac15241e37514e78a0a9979a293b3.ssl.cf1.rackcdn.com/377/c1e29d293735d59fc5b0b7eda470ede94b70312f/check/ansible-test-cloud-integration-aws-py36_0/6be4952/job-output.txt
```
2021-09-05 09:32:53.579799 | About to run: ansible-test integration --diff --no-temp-workdir --skip-tags False --retry-on-error --continue-on-error --python 3.6 -vvvv ['elb_classic_lb']
...
2021-09-05 09:32:54.651353 | fedora-34 | ERROR: Target pattern not matched: [elb_classic_lb]
```

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
recheck

**Review by alinabuzachis — APPROVED** [review]
All this seems good to me! Thank you @tremble, the review took more than I expected.

File: plugins/modules/elb_classic_lb.py [hunk 1]
```diff
@@ -0,0 +1,2156 @@
...
+
+
+DOCUMENTATION = '''
+---
+module: elb_classic_lb
+version_added: 1.0.0
+description:
+  - Creates, updates or destroys an Amazon Elastic Load Balancer (ELB).
+  - This module was renamed from M(ec2_elb_lb) to M(elb_classic_lb) in version
+    2.0.0 of the amazon.aws collection.
```
Comment (tremble) [inline plugins/modules/elb_classic_lb.py:16]:
```suggestion
    2.1.0 of the amazon.aws collection.
```

File: plugins/modules/elb_classic_lb.py [hunk 2]
```diff
@@ -0,0 +1,2156 @@
...
+          - If this parameter is not specified, the stickiness session lasts for the duration of the browser session.
+          - Ignored if I(enabled=false).
+        required: false
+        type: int
+  wait:
+    description:
+      - When creating, deleting, or adding instances to an ELB, if I(wait=true)
+        Ansible will wait for both the load balancer and related network interfaces
+        to finish creating/deleting.
+      - Support for waiting when adding instances was added in release 2.0.0.
```
Comment (tremble) [inline plugins/modules/elb_classic_lb.py:269]:
```suggestion
      - Support for waiting when adding instances was added in release 2.1.0.
```

File: plugins/modules/elb_classic_lb.py [hunk 3]
```diff
@@ -0,0 +1,2156 @@
...
+      - A dictionary of tags to apply to the ELB.
+      - To delete all tags supply an empty dict (C({})) and set
+        I(purge_tags=true).
+    type: dict
+  purge_tags:
+    description:
+      - Whether to remove existing tags that aren't passed in the I(tags) parameter.
+    type: bool
+    default: true
+    version_added: 2.0.0
```
Comment (tremble) [inline plugins/modules/elb_classic_lb.py:289]:
```suggestion
    version_added: 2.1.0
```

**Pull request merged**